modules/renter/contractor/formcontract.go

```diff
@@ -382,7 +128,7 @@ func (c *Contractor) formContracts(a modules.Allowance) error {
 		c.log.Printf("WARN: failed to form desired number of contracts (wanted %v, got %v): %v", a.Hosts, numContracts, strings.Join(errs, "\n"))
 	}
 	c.mu.Lock()
-	c.renewHeight = endHeight
+	c.renewHeight = c.blockHeight + a.Period // TODO: this risks desync
```

Comment:
What needs to happen so that desync is no longer risked? I feel like this is a pretty important problem.